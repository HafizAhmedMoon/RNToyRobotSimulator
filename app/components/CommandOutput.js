import PropTypes from 'prop-types';
import React from 'react';
import { StyleSheet, Text, View } from "react-native";

export class CommandOutput extends React.Component {

  static propTypes = {
    output: PropTypes.string.isRequired
  };

  render() {
    const {output} = this.props;
    return (
      <View style={styles.container}>
        <Text>{output}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    alignSelf: 'stretch',
    paddingHorizontal: 10
  },
});
