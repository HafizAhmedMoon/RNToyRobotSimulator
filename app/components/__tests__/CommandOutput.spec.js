import React from 'react';
import { shallow } from 'enzyme';
import { CommandOutput } from '../CommandOutput';

describe("CommandOutput Component", () => {

  it('renders as expected', () => {
    const wrapper = shallow(
      <CommandOutput output=""/>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a Text', () => {
    const wrapper = shallow(
      <CommandOutput output="Some output"/>
    );

    const Text = wrapper.find('Text');
    expect(Text).toHaveLength(1);
  });

});
