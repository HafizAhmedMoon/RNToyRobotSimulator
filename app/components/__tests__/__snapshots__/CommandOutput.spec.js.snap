// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandOutput Component renders as expected 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "paddingHorizontal": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  />
</View>
`;
